### ISSUE TITLE
Missing imports in pod files

### ISSUE THREAD

**Issue by mrdavidrees** [vimeo/VimeoUpload#92]
Most of the files within the VimeoUpload pod only have one or two imports of Foundation or UIKit, none of them contain imports of AFNetworking, Or VIMNetworking. What Am I missing here? How has this compiled for your team? Most files have an error where it's missing an import.

**Comment by alfiehanssen** [comment]
Hey @mrdavidrees, great question! Unfortunate answer: 

The Vimeo iOS app supports iOS7, for this reason we cannot leverage dynamic Swift frameworks in the project. This means we have to include the VimeoUpload raw source in that project instead of consuming it as a pod. And this means we cannot add the import statements where necessary because our iOS7 project wouldn't compile...yet. 

We will likely drop support for iOS7 sometime in the next 2-3 months. Which means we'll be able to modify VimeoUpload in the necessary ways. 

In the meantime, check out the `CAM-205` [pull request](https://github.com/vimeo/VimeoUpload/pull/85) and corresponding [branch](https://github.com/vimeo/VimeoUpload/tree/CAM-205) of VimeoUpload. This is set up to be consumed as a development pod and contains the import statements. We set this branch up so that we can consume it in our Cameo-iOS project which is iOS8+. It's working well although there may be some kinks to work out. 

Note however: the `CAM-205` branch depends on our new Swift networking iOS SDK (we're deprecating the Objc VIMNetworking and replacing it with the Swift VimeoNetworking). VimeoNetworking is private at the moment because it's under active development. However, no reason to not make it public now so you and others can work with the `CAM-205` branch. 

In order to make VimeoNetworking public we will need to add a note to the README saying it's under active development etc (although it's nearly finished). We'll try to do this by the end of the day today. Ping us if you don't have access to VimeoNetworking by the week's end. 

Short answer: 

You have two options. (1) Include the VimeoUpload raw source in your project or (2) consume the `CAM-205` branch as a development pod once we make VimeoNetworking public (this week). 

Hope this helps.

**Comment by alfiehanssen** [comment]
@mrdavidrees, VimeoNetworking is now public: 

[https://github.com/vimeo/VimeoNetworking](https://github.com/vimeo/VimeoNetworking)

**Comment by mrdavidrees** [comment]
@alfiehanssen 
I would love to include the new VimeoNetworking api in the app. Thanks so much for your help, i'm excited to use the new code, I'll let you know of any plausible bugs when using the API.

**Comment by mrdavidrees** [comment]
Hey I've spent most of the day figuring out the library. I've got a few questions of things I couldn't really figure out of my own, also a few errors I encountered on the way.

**Issues**
executed default case when debugging, possible timeout, authentication unsecessful
breakpoint was at line 248, AuthenicationController handleAuthenticationResult(:_) ->

    private func errorForStatus(status: OSStatus) -> NSError
    default:
                errorMessage = "undefined error"


in compiled swift generated interface file for VimeoUpload framework (VimeoNetworking-Swift.h), have to remove this before can compile

    Requires #import <Foundation/Foundation.h>  (file not found)



**Questions**
Our plan is to allow app users to upload their personal video to our Vimeo account
These videos can be viewed by anyone logged into app.

Get authenticated user from AuthenticationController

	let scopes = [Scope.Private, Scope.Public, Scope.Create, Scope.Upload]
        let config = AppConfiguration(clientKey: “XXXX”, clientSecret: “XXX/XXXX/XXX”, scopes: scopes)
        let client = VimeoClient(appConfiguration: config)
        
        let authentication = AuthenticationController(client: client)
        authentication.clientCredentialsGrant { result in

            switch result {
            case .Failure(let error):
                print(error)
                
            case.Success(let client):
                
                print(client.isAuthenticated())
            }   
        }

It's mentioned in VimeoUpload that the 'New way' would only require creating a task (Upload Descriptor) and Adding it to the VimeoUploader, but if is this still the case if a media item is in iCloud?

Also can I export a smaller resized video in the middle of fetching an item from iCloud or Device, and uploading it to Vimeo?

like in the completion block of PHAssetCloudExportQuotaCreateOperation, create a new task for resizing, and on the completion of that, call VideoUploader.uploadVideo(:_), is this an ok solution? I’m also assuming if the iCloud operation fails it’s not going to re-start like the video upload operation

This is what i'd imagine it'd look like

	let session = VimeoSessionManager.backgroundSessionManager(identifier: self.dynamicType.BackgroundIdentifier, appConfiguration: config)
	let asset = story.assets[0]
        let user = Vimeo.sharedInstance.user!
        let session = Vimeo.sharedInstance.session!
        let operation = PHAssetCloudExportQuotaCreateOperation(me: user, phAsset: asset, sessionManager: session)
        operation.completionBlock = {

            let url = operation.url!
            let resize = VideoResizeOperation(url: url)
            //Resize Video
            
            
            //Resize Video Completion
            resize.completionBlock = {

                let uploadTicket = operation.uploadTicket!
                let descriptor = UploadDescriptor(url: url, uploadTicket: uploadTicket)
                descriptor.identifier = asset.localIdentifier
                
                Vimeo.sharedInstance.uploadVideo(descriptor: descriptor)
            }
        }

**Comment by mrdavidrees** [comment]
@alfiehanssen Any ideas on the second issue above? Causing some strife with our autobuild..

**Comment by alfiehanssen** [comment]
Hey @mrdavidrees, apologies for the delay. 

1. @huebnerob Can you speak to on "Issue 1" described above in the previous comment? @mrdavidrees Can you provide some additional detail about (1) What authentication method you're using, (2) any other details that would help provide context?

2. Different import statement issue but perhaps [this PR](https://github.com/vimeo/VimeoNetworking/pull/20) had a positive impact on the framework import issue you describe? @huebnerob Any ideas on this one? Sounds like the exact same issue but for a different framework. 

3. Regarding uploading videos that reside in iCloud. Regardless of whether you're using the "new" or "old" upload system, you'll still have to use one of the NSOperation subclasses to download the file to the device and then export a copy of it before uploading. Once the download + export is complete you would construct your upload descriptor and add it to the manager. We opted specifically to not include download and export "inside" the descriptor system (1) because these are operations that must occur in the foreground and (2) it gives the library user the flexibility to design for whatever user experience they'd like. 

4. "Can I export a smaller resized video in the middle of fetching an item from iCloud or Device, and uploading it to Vimeo?" You can change the size of the video you export by adjusting the `exportPreset` value leveraged by the AVAssetExportSession. If you're asking whether you can perform the export **at the same time** as the download, I doubt it. Especially for iCloud download. If you're downloading from a server you control and therefore have greater access to the file you're downloading to, you could probably use an AVAssetReader/Writer to do something like what you're describing. But I doubt it's possible with iCloud download because your access / permissions to the local file are limited. 

5. If you simply want to resize the video once iCloud download is complete, this is definitely possible. You would probably want to (1) download the video, (2) either (a) export the video with a preset of your choice or (b) create a custom export session to export at the size you want. 

6. The NSOperation subclasses that handle download and export will not auto-restart. It's up to you to detect failure and restart. 

7. The code snippet you stub out to do download > export > resize > upload looks good.

🎉 

Let us know if you have other questions
